core/src/main/java/org/infinispan/container/DataContainer.java

```diff
@@ -113,7 +112,6 @@ default int size() {
    /**
     * Removes all entries in the container
     */
-   @Stop(priority = 999)
```

Comment:
I decided to stop allowing lifecycle methods in interfaces, because we don't have any ComponentAccessor for interfaces, and it complicates ordering. Thinking more about it, I should probably also prohibit lifecycle annotations on methods that already have the annotation in the super class, so it's clear that they're invoked through the super class accessor.

Reply:
Updated to prohibit lifecycle annotations on overridden methods

Reply:
Sounds good to me, just to be sure these types of checks are done at compile time ?

Reply:
Yes, it's all in the annotation processor. Currently core doesn't read the annotations at runtime, and the server doesn't include the annotations jar at all.